app/grandchallenge/reader_studies/views.py

```diff
@@ -1434,12 +1434,14 @@ def create_civ_for_value(
         ci = ComponentInterface.objects.get(slug=ci_slug)
         current_value = instance.values.filter(interface=ci).first()
         if ci.is_json_kind and not ci.requires_file:
-            if current_value:
+            if current_value and current_value.value != new_value:
                 assigned_civs.append(current_value)
-            val = ComponentInterfaceValue.objects.create(
-                interface=ci, value=new_value
-            )
-            instance.values.add(val)
+            if new_value and current_value.value != new_value:
```

Comment:
Can these three if statements be tidied up? Quite a lot going on there and some duplication.